sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -981,3 +981,24 @@ def test_init():
                            max_iter=1, random_state=random_state).fit(X)
 
     assert_greater(gmm2.lower_bound_, gmm1.lower_bound_)
+
+def test_init_param():
+    # Check that the init_param options produces valid output.
+    # Compare all to default - kmeans.
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng)
+
+    n_components, n_features = rand_data.n_components, rand_data.n_features
+    X = rand_data.X['full']
+    gmm = GaussianMixture(n_components=n_components, random_state=rng,
+                          init_params='kmeans')
+    gmm.fit(X)
+    default_means = gmm.means_
+
+    INIT_TYPE = ['random','rand_data','k-means++']
+
+    for init in INIT_TYPE:
+        gmm = GaussianMixture(n_components=n_components, random_state=rng,
+                              init_params=init, tol=1e-06, max_iter=1000)
```

Comment:
Can we check that the different `init_params` have a different effect with the same `random_state`, if only to make sure they're being passed? Or can we check the distribution of the means with `max_iter=1`?

Reply:
Getting the initial means out could be a bit hard from the way the classes are constructed but I think you're right that with `max_iter=1` they should still be unique enough. The only ones that might be close would be `k-means` and `k-means++` but I will take a look.

Also happy to take a look at the user guide after I've adjusted this.